src/components/tree-item/tree-item.scss

```diff
@@ -5,6 +5,10 @@
     cursor-pointer;
 }
 
+[hidden] {
+  @apply hidden;
+}
```

Comment:
Do we need this? `hidden` is a global attribute so I think it should just work (famous last words)

Reply:
I remember having this discussion before with others and as it turns out even though you'd expect the global attribute "hidden" to have that behavior by default, it doesn't and needs to be set.

Reply:
> even though you'd expect the global attribute "hidden" to have that behavior by default, it doesn't and needs to be set.

Natively, `hidden` does hide the element, but in this case, we need an attribute selector so it can have a higher specificity and override the `display` prop set in the [`.actions-end` style here](https://github.com/Esri/calcite-components/pull/6005/files#diff-c7e860104581dc1e9bebfb6a5733bb56066510cf1e9abfb39102bbecec90ec3dR110) ([example](https://codepen.io/jcfranco/pen/abjYOoL)).

Sidebar: maybe we need to add a selector for `hidden` within our component's shadow DOM to [`includes.scss`](https://github.com/Esri/calcite-components/blob/master/src/assets/styles/includes.scss#L18-L20).